ambry-router/src/test/java/com.github.ambry.router/InMemoryRouter.java

```diff
@@ -157,7 +157,11 @@ public ByteBuffer getBlob(ByteRange range) throws RouterException {
         } catch (IllegalArgumentException e) {
           throw new RouterException("Invalid range for blob", e, RouterErrorCode.RangeNotSatisfiable);
         }
-        buf = ByteBuffer.wrap(blob.array(), (int) resolvedRange.getStartOffset(), (int) resolvedRange.getRangeSize());
+        byte[] bytes = new byte[(int) resolvedRange.getRangeSize()];
+        ByteBuffer duplicate = blob.duplicate();
+        duplicate.position((int) resolvedRange.getStartOffset());
+        duplicate.get(bytes);
+        buf = ByteBuffer.wrap(bytes);
       }
```

Comment:
I have seen a really weird bug, where a range request would somehow pollute the blob data here. I change the way we are dealing with range request data to copying it out. And it fixes the problem. This is weird, but I won't get into too much detail.